app/services/search/school_search.rb

```diff
@@ -113,4 +108,16 @@ def apply_organisation_type_filter(scope)
 
     scope.where(school_type: selected_school_types)
   end
+
+  def apply_school_type_filter(scope)
+    return scope unless school_types.present?
+
+    if school_types.include?("special_school") && school_types.include?("faith_school")
+      return scope.where.not("gias_data ->> 'ReligiousCharacter (name)' IN (?)", Organisation::NON_FAITH_RELIGIOUS_CHARACTER_TYPES).or(scope.where(school_type: Organisation::SPECIAL_SCHOOL_TYPES))
+    end
+
+    return scope.where(school_type: Organisation::SPECIAL_SCHOOL_TYPES) if school_types.include?("special_school")
+
+    scope.where.not("gias_data ->> 'ReligiousCharacter (name)' IN (?)", Organisation::NON_FAITH_RELIGIOUS_CHARACTER_TYPES)
+  end
```

Comment:
Similar to above on the other search filtering, I find guard clauses here making the code harder to read (besides the first one) than explicit branching based on the school_types values.

Again this is subjective and I propose what I found easier to read/follow so you can compare and decide :). 
```suggestion
def apply_school_type_filter(scope)
  return scope unless school_types.present?

  if school_types.include?("special_school") && school_types.include?("faith_school")
    scope.where.not("gias_data ->> 'ReligiousCharacter (name)' IN (?)", Organisation::NON_FAITH_RELIGIOUS_CHARACTER_TYPES)
         .or(scope.where(school_type: Organisation::SPECIAL_SCHOOL_TYPES))
  elsif school_types.include?("special_school")
    scope.where(school_type: Organisation::SPECIAL_SCHOOL_TYPES)
  elsif school_types.include?("faith_school")
    scope.where.not("gias_data ->> 'ReligiousCharacter (name)' IN (?)", Organisation::NON_FAITH_RELIGIOUS_CHARACTER_TYPES)
  else
    scope
  end
end
```